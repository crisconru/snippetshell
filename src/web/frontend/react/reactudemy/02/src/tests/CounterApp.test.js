import React from 'react';
//import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp'


describe('Pruebas CounterApp', () => {
    /*
    test('FirstApp debe devolver "Hola mundo zanguango"', () => {
        const resultado = 'Hola mundo zanguango'
        const col = 'zanguango'
        const { getByText } = render( <FirstApp coletilla={col}/> )
        expect( getByText(resultado) ).toBeInTheDocument()
    })
    */

   let componente = shallow( <CounterApp /> )

    beforeEach(() => {
        componente = shallow( <CounterApp /> )
    })

    test('Debe mostrar <CounterApp/> con 10', () => {
        //const value = parseInt( componente.find('h2').text() )
        expect(componente).toMatchSnapshot()
        //expect(value).toBe(10)
    })
    test('Debe mostrar <CounterApp/> con 100', () => {
        const val = 100
        const componente = shallow( <CounterApp value={val}/> )
        const value = parseInt( componente.find('h2').text() )
        expect(componente).toMatchSnapshot()
        expect(value).toBe(val)
    })

    test('Boton incrementar en 1', () => {
        componente.find('button').at(0).simulate('click')
        const val = componente.find('h2').text().trim()
        expect(val).toBe('11')
    })

    test('Boton decrementar en 1', () => {
        componente.find('button').at(2).simulate('click')
        const val = componente.find('h2').text().trim()
        expect(val).toBe('9')
    })

    test('Boton resetear a 10', () => {
        const valor = 100
        const componente = shallow( <CounterApp value={valor}/> )
        componente.find('button').at(0).simulate('click')
        componente.find('button').at(1).simulate('click')
        const val = componente.find('h2').text().trim()
        expect(val).toBe(`${valor}`)
    })


})