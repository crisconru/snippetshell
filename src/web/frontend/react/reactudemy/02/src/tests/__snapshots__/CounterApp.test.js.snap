// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas CounterApp Debe mostrar <CounterApp/> con 10 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`Pruebas CounterApp Debe mostrar <CounterApp/> con 100 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
